### PR TITLE
add filters for irrelevant files in ISISDATA

### PR DESCRIPTION
**Pull request by antonhibl**
## Description
added filters for the following files in the ISISDATA download script, these are being imminently removed:

/a_older_versions/*
/kernels/*/former_versions/
/corrupt_files/*
/zzarchive/**
/kernels/*/original
gallileo/kernels/ck/prime_mission//extended_mission/
gallileo/kernels/*/prime_mission/
galileo/kernels/ck/GEM/
clementine1/kernels/ck/save
chandrayaan1/kernels/spk/SAVE_SCS_2017-11-22
/kernels/fk/release./*
cassini/kernels/fk/Archive

## Related Issue
Data storage concerns in ISISDATA area

## How Has This Been Validated?
tested downloadIsisData script on my machine and it works fine with these changes applied

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Infrastructure change (changes to things like CI or the build system that do not impact users)

## Checklist:
- [x] I have read and agree to abide by the [Code of Conduct](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/Code-Of-Conduct.md)
- [x] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [x] I have added myself to the [.zenodo.json](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/.zenodo.json) document.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [x] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

File: isis/config/exclude_list.txt
Comment (Kelvinrr) [inline isis/config/exclude_list.txt]: Just to repeat here, might want to put this as a list in the script so users don't have to download another file if they are just wget-ing it.

File: isis/scripts/downloadIsisData
Comment (Kelvinrr) [inline isis/scripts/downloadIsisData]: What would happen if the user inputs their own exclude list?

File: isis/pytests/test_IsisDownload.py
Comment (Kelvinrr) [inline isis/pytests/test_IsisDownload.py]: If you are sourcing the script, you can probably use the same exclude list from the actual script rather than duplicating here.

File: isis/scripts/downloadIsisData
Comment (Kelvinrr) [inline isis/scripts/downloadIsisData]: You might want to check for additional --include or --filter flags passed into by the user and merge them since rclone docs say mixing them can cause weird behavior. Maybe default to --filter, and grab any user --include and --exclude flags and merge them into the filter flag?

**Review by Kelvinrr — COMMENTED** [review]
test looks good 👍🏽 just one major hurdle left.

File: isis/scripts/downloadIsisData
Comment (Kelvinrr) [inline isis/scripts/downloadIsisData]: Would this actually filter them out? From the docs they need a + or - sign in front to explicitly include or exclude from the files as --filter is basically a generic version of include and exclude.

File: isis/scripts/downloadIsisData
Comment (Kelvinrr) [inline isis/scripts/downloadIsisData]: Would this be robust enough? So capture `--include "stuff"`, `--include="stuff"`, and (I think this is valid too) `--include"stuff"`.

File: isis/scripts/downloadIsisData
Comment (Kelvinrr) [inline isis/scripts/downloadIsisData]: From the docs: https://rclone.org/filtering/

`--filter +` doesn't not import exclude `*` like include does. If the user specifies --include. If there is an --include flag, we might want to add `- *` at the bottom to simulate this? Remember, the goal is to honor the user's intention when they input their own flags, if they explicitly want the stuff we are filtering, we should be able to make that work naturally. 

Also, this doesn't seem to be accounting for the "-" or "+" in --filter to specify if it's including or excluding something? I would say read those docs I linked to get an idea of how to make this work.

File: isis/scripts/downloadIsisData
Comment (Kelvinrr) [inline isis/scripts/downloadIsisData]: Looking at this, it still looks like the flag isn't being set correctly. remember, from the docs I linked, that regexes that need to be includes need to be prefixed with "+" and things that are to be ignored with "-" for every regex. These basically have to be added as there own --filter flag. So you can't use the "{-x,- y, +z }" syntax. e.g `rclone --filter "- gallileo/kernels/ck/prime_mission/*/extended_mission/*" --filter "- /a_older_versions/* ..."`

also, remember if the user passes in a --include, you need to append `--filter "- *"`, in order to replicate the functionality of --include

**Comment by Kelvinrr** [comment]
tested this locally and I am still getting some of the kernels that are supposed to be filtered

**Comment by antonhibl** [comment]
@Kelvinrr shoot, what files were you seeing not filtering? I was testing by downloading the galileo area and it seemed to work for those files for me; specifically galileo/kernels/ck/GEM/ was filtered when I tested on my machine.

**Comment by Kelvinrr** [comment]
It should get everything under extended mission and those things were still being included when downloading Galileo.

File: isis/scripts/downloadIsisData
Comment (Kelvinrr) [inline isis/scripts/downloadIsisData]: I think this is the problem, there are two filters mashed into one. Also the asterisks seem to be messing up the filters. I guess rclone doesn't wants either double asterisks or a substring. 


This seemed to be working for me. Added a few things on top of the existing list. 

```python 
exclude_list = [
        'a_older_versions/',
        'former_versions/',
        'corrupt_files/',
        'zzarchive/',
        'original/',
        'prime_mission/',
        'extended_mission/',
        'prime_mission/',
        'GEM/',
        'save/',
        'SAVE_SCS_2017-11-22/',
        'release*/',
        'Archive/',
        'ek/',
        '*.lbl',
        'unitVectors/',
        '*.pdf'
    ]
```

**Comment by Kelvinrr** [comment]
testing filter also failed: 

```shell
(isisdev23) IGSWZAWGLT00017:scripts krodriguez$ ./downloadIsisData galileo /tmp/isisdata -vv -n5 --include "**/ck/*"
Traceback (most recent call last):
  File "/Users/krodriguez/repos/isis/isis/scripts/./downloadIsisData", line 51, in <module>
    include_argers = [arg for arg in sys.argv if arg.startswith("--include")][0].split("=")[1]
IndexError: list index out of range
```

as I said in a previous comment, user can input `--include "filter*` or `--include="filter"`. Why parse the arg yourself when argparse can do it for you?

**Comment by Kelvinrr** [comment]
I think this is good, but we should wait for #5111 to merge before we merge this so we can get good tests for any changes to the download script using that new application.

**Review by Kelvinrr — DISMISSED** [review]
Im just gonna approve this so we can shrink the data area already.

**Comment by AustinSanders** [comment]
@Kelvinrr @antonhibl Is this ready to be merged?

**Comment by antonhibl** [comment]
@AustinSanders I think so but would double check with Kelvin as he was still working on this afaik @Kelvinrr

**Comment by antonhibl** [comment]
In the future adding the ability to have mission specific exclude/include patterns and lists be applied depending what base area is specified would be very useful as it would allow for more granularity in filtering and possibly better mission by mission optimizations and speed-ups.

**Review by AustinSanders — APPROVED** [review]

**Pull request merged**
